src/pages/recipes/events/index.md

```diff
@@ -0,0 +1,129 @@
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+let myEventListener = app.addEventListener("afterNew", myDisplayEventType);
+
+function myDisplayEventType(myEvent){
+    console.log(This event is the ${myEvent.eventType event.);
+}
+```
+
+**Removing an eventListener**
+
+To remove the eventListener created by the preceding script, run the following script:
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+let myResult = app.removeEventListener("afterNew", myDisplayEventType);
+```
+
+An event can trigger multiple eventListeners as it propagates through the scripting object model. The following sample script demonstrates an event triggering eventListeners registered to different objects:
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+let myApplicationEventListener = app.eventListeners.add("beforeImport", myEventInfo);
+let myDocumentEventListener = app.documents.item(0).eventListeners.add("beforeImport", myEventInfo);
+function myEventInfo(myEvent){
+    let myString = "Current Target: " + myEvent.currentTarget.name;
+    console.log(myString);
+}
+```
+When you run the preceding script and place a file, InDesign displays following information in sequence, and on the console:
+1. Document name
+2. Application name
+
+Use the following script to remove the event listeners added by the preceding script:
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+app.removeEventListener("beforeImport", myEventInfo);
+app.documents.item(0).removeEventListener("beforeImport", myEventInfo);
+```
+
+## Writing events script to run through out the session
+After the event listener is associated, the script is blocked by a promise and wait for the event in the background until the event listener is removed. If the event occurs, there is an automatic callback to execute the related function. If the event listener is removed, the promise can be resolved to complete the script execution.
+
+Here is a sample events script to run and listen for “afterNew” event throughout the InDesign session and prints a relevant string whenever a new document is created.
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+ 
+await listenAfterNew();
+ 
+async function listenAfterNew(){
+  return new Promise((resolve, reject) => { mySnippet(); })
+}
+ 
+function mySnippet(){
+    //![Add event listener.]
+    let myEventListener = app.addEventListener("afterNew", myDisplayEventType);
+    //![Add event listener.]
+}
+ 
+//![Add event listener - functions.]
+function myDisplayEventType(myEvent){
+    console.log("This event is the " + myEvent.eventType + " event.");
+}
+```
+Here is a sample events script to run and listen for "afterNew" event only once and remove, which prints a relevant string if a new document is created.
```

Comment:
please highlight the text 'only once' by bolding it.

Reply:
Done